pom.xml

```diff
@@ -199,6 +199,37 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>flatten-maven-plugin</artifactId>
+                <version>1.2.7</version>
+                <configuration>
+                    <updatePomFile>true</updatePomFile>
+                    <pomElements>
```

Comment:
There is a property called flattenMode.

https://www.mojohaus.org/flatten-maven-plugin/usage.html

https://github.com/mojohaus/flatten-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/flatten/FlattenMode.java

Is it possible to use it instead of defining all the pom elements?

Reply:
After reading about it I believe we should use flattenMode "bom" (Bill of materials) which keeps all optional POM elements that are required for OSS Repository-Hosting and additionally dependencyManagement and properties values.
Thanks Anastasiia.